Modules/CityWeatherEntriesVC/CityWeatherEntriesVC.swift

```diff
@@ -0,0 +1,207 @@
+//
+//  CityWeatherEntriesVC.swift
+//  WeatherDemoApp
+//
+//  Created by ahmed on 05/04/2024.
+//
+
+
+import UIKit
+
+final class CityWeatherEntriesVC: UIViewController {
+    
+    let cityWeatherInfo: LocalStorageCity
+    private var gradientView: CAGradientLayer?
+    
+    init(cityWeatherInfo: LocalStorageCity) {
+        self.cityWeatherInfo = cityWeatherInfo
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+ 
+    // MARK: Life Cycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        addGradient()
+        addBottomImageConstaints()
+        addPageTitleLabelConstaints()
+        addBtnViewConstaints()
+        
+        setupPageTitleLbl()
+        setupLeftBtn()
+        addSearchResultsTableConstaints()
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        // to activate gradient
+        gradientView?.frame = view.bounds
+    }
+    
+    // MARK: View Configurators
+    private func addGradient() {
+        let pageGradient = Colors.pageGradient
+        
+        gradientView = view.setGradient(colors: pageGradient.colors,
+                                        locations: pageGradient.locations,
+                                        startPoint: pageGradient.startPoint,
+                                        endPoint: pageGradient.endPoint)
+    }
+    
+    private func setupLeftBtn() {
+        leftBtn.setCornerRadius(14)
+        
+        /// Action
+        let addBtnAction: VoidCallback = { [unowned self] in
+            navigationController?.popViewController(animated: true)
+        }
+        
+        let model = ReusableBtnModel(btnTappedAction: addBtnAction,
+                                     btnImage: UIImage(systemName: "arrow.left"))
+        leftBtn.configureBtnWith(model)
+    }
+    
+   
+    private func setupPageTitleLbl() {
+        let titleTxt = "\(cityWeatherInfo.cityName) \n Historical"
+        pageTitleLbl.setTitle(titleTxt)
+    }
+    
+    // MARK: SubViews
+    private lazy var pageTitleLbl = ReusableBoldLabel()
+    private lazy var leftBtn = ReusableButton()
```

Comment:
```suggestion
    private let titleLabel = ReusableBoldLabel()
    private let trailingButton = ReusableButton()
```
Why you're not using the navigation bar?